### PR TITLE
Update bip32 to v2 to support Node.js v12

### PR DESCRIPTION
**Pull request by vanesyan**

### CODE REVIEW

**Comment by vanesyan** [comment]
Can you also help me with circle-ci setup to test module also on Node.js v12

**Comment by faboweb** [comment]
> Can you also help me with circle-ci setup to test module also on Node.js v12 please?

How can I help?

I guess if you update the used image you can test on any Node.js version you like: https://github.com/luniehq/cosmos-keys/blob/develop/.circleci/config.yml#L39

File: src/cosmos-keys.ts [hunk 1]
```diff
@@ -72,9 +72,14 @@ function deriveMasterKey(mnemonic: string): bip32.BIP32 {
   return masterKey
 }
 
-function deriveKeypair(masterKey: bip32.BIP32): KeyPair {
+function deriveKeypair(masterKey: bip32.BIP32Interface): KeyPair {
   const cosmosHD = masterKey.derivePath(hdPathAtom)
   const privateKey = cosmosHD.privateKey
+
+  if (!privateKey) {
```
Comment (faboweb) [inline src/cosmos-keys.ts]: Is there any case were this would happen? Does `derivePath` return `{}` in any case.

**Comment by vanesyan** [comment]
> How can I help?

I mean, should I just update Node.js version, or need to setup matrix of versions, like test both on Node.js v10 and v12

File: src/cosmos-keys.ts (hunk 1 above)
Comment (vanesyan) [reply src/cosmos-keys.ts]: No. `privateKey` has typing `Buffer | undefined`, so we have to check if it really exists. But I think we can safely cast it to `Buffer` as `privateKey` is always created when using `fromSeed`

File: package.json [hunk 2]
```diff
@@ -89,8 +89,6 @@
   },
   "devDependencies": {
     "@types/bech32": "^1.1.1",
-    "@types/bip32": "^2.0.0",
```
Comment (vanesyan) [inline package.json]: Types are redundant as they are already bundled by bip32 and bip39 packages on their own

File: src/cosmos-keys.ts (hunk 1 above)
Comment (faboweb) [reply src/cosmos-keys.ts]: The latter sounds like a good idea. The current implementation suggests it is undefined in some case.

**Comment by faboweb** [comment]
> I mean, should I just update Node.js version, or need to setup matrix of versions, like test both on Node.js v10 and v12

No testing on a late Node.js version should be fine.

**Comment by faboweb** [comment]
I want to add a huge thank you for providing this PR!

File: src/cosmos-keys.ts (hunk 1 above)
Comment (vanesyan) [reply src/cosmos-keys.ts]: Updated, can you take a look when you will have time, ty

**Comment by faboweb** [comment]
Somehow the tests are not working. We need to investigate. I am sorry for the holdup.

File: package.json [hunk 3]
```diff
@@ -89,8 +89,6 @@
   },
   "devDependencies": {
     "@types/bech32": "^1.1.1",
-    "@types/bip32": "^1.0.2",
```
Comment (jbibla) [inline package.json]: how come these are not needed as dev dependencies anymore?

Comment (vanesyan) [reply package.json]: look https://www.npmjs.com/package/@types/bip32

Comment (jbibla) [reply package.json]: say no more!

**Pull request merged**